### DISCUSSION TITLE
"window-decoration: true" not working anymore when "gtk-titlebar: false" - titlebar is not displayed and new tab crashes Ghostty

### DISCUSSION THREAD

**Discussion by ghost** [ghostty-org/ghostty#4172] [Feature Requests, Ideas] [answered]
With Ghostty 1.0
```
      gtk-titlebar: false
      window-decoration: true
```
  - Titlebar was shown on launch and could be toggled hidden with toggle_window_decorations, as documented and expected.

Currently:
  - Titlebar is NOT shown on launch and can NOT be toggled to display with toggle_window_decorations.
  - Creating a new tab using shft-ctrl-T crashes Ghostty.

**Comment by jcollie** [comment]
>  as documented and expected.

I would differ in your interpretation of the documentation and expectation. When `gtk-titlebar=false` I would expect the titlebar _not_ to show up, regardless of the value of `window-decoration`. There are other forms of window decoration like a drop shadow and rounded corners which someone may want to keep while hiding the titlebar.

**Comment by jcollie** [comment]
> Creating a new tab using shft-ctrl-T crashes Ghostty.

I am unable to reproduce, and this is unrelated to the interaction between `gtk-titlebar` and `window-decoration` so please start a new discussion regarding this. Please include more information like the output of `ghostty +version` and any logging or tracebacks that show up when you run Ghostty from the CLI of another shell or instance.

**Comment by ghost** [comment, marked as answer]
> I would differ in your interpretation of the documentation and expectation. When gtk-titlebar=false I would expect the titlebar not to show up, regardless of the value of window-decoration. There are other forms of window decoration like a drop shadow and rounded corners which someone may want to keep while hiding the titlebar.

Modifying the explanation of the documentation to obscure a clearly introduced new bug is not the way to go.
I was planning on adding Ghostty to the official package repository of the Artix Linux distribution, but now it absolutely fails to qualify.
